integration/unplugin/README.md

```diff
@@ -0,0 +1,92 @@
+# unplugin-civet
+
+Use Civet in your projects with Vite, Webpack, Rspack, Rollup and esbuild, with `dts` generation supported.
+
+## Usage
+
+The only setup required is to install the plugin and adding it your bundler's config:
+
+```bash
+npm i -D unplugin-civet
+```
+
+### Vite
+
+```ts
+// vite.config.ts
+import { defineConfig } from 'vite';
+import civetVitePlugin from 'unplugin-civet/vite';
+
+export default defineConfig({
+  // ...
+  plugins: [
+    civetVitePlugin({
+      // options
+    }),
+  ],
+});
+```
+
+### Rollup
+
+```ts
+// rollup.config.ts
+import civetRollupPlugin from 'unplugin-civet/rollup';
+
+export default {
+  // ...
+  plugins: [
+    civetRollupPlugin({
+      // options
+    }),
+  ],
+};
+```
+
+### ESBuild
+
+```ts
+import esbuild from 'esbuild';
+import civetEsbuildPlugin from 'unplugin-civet/esbuild';
+
+esbuild
+  .build({
+    // ...
+    plugins: [civetEsbuildPlugin()],
+  })
+  .catch(() => process.exit(1));
+```
+
+### Webpack
+
+```js
+const civetWebpackPlugin = require('unplugin-civet/webpack');
+
+module.exports = {
+  // ...
+  plugins: [
+    civetWebpackPlugin({
+      // options
+    }),
+  ],
+};
+```
+
+## Options
+
+```ts
+interface PluginOptions {
+  dts?: boolean;
+  outputExtension?: string;
+  js?: boolean;
+  transformOutput?: (
+    code: string,
+    id: string
+  ) => TransformResult | Promise<TransformResult>;
+}
+```
+
+- `dts`: `unplugin-civet` also supports generating `.d.ts` type definition files from the civet source, which is useful for building libraries.
+- `outputExtension`: Output filename extension to use. Default: `.civet.jsx`, or `.civet.tsx` if `js` is `false`.
+- `js`: Whether to transpile to JS or TS.
```

Comment:
A default of `js: true` would break `dts: true`. We could make `dts` imply `js: false`, but honestly `js: false` is safer; some TypeScript features are not correctly implemented by Civet with `js: true`, and it just costs a bit of time to use already existing TS integration. So I think `js: false` is best.

Reply:
We could give imply `js` based on the value of `dts` first,
```js
const transpileToJS = options.dts || options.js;
```
in which case both options should work. My main concern was that rollup and webpack can't process typescript without additional plugins, so would need to set `js: true` to make the setup work, but if `js: false` has better typescript features then let's go with that default.

Reply:
Maybe we can include the `{js: true}` in the README where we give recommended workflows for Webpack and Rollup?